python/tests/transform_coords_test.py

```diff
@@ -0,0 +1,342 @@
+# Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+import pytest
+import numpy as np
+import scipp as sc
+
+
+def _make_xy():
+    Nx = 40
+    Ny = 60
+    return sc.DataArray(data=sc.ones(dims=['x', 'y'], shape=[Nx, Ny]),
+                        coords={
+                            'x':
+                            sc.linspace(dim='x',
+                                        unit='m',
+                                        start=0.1,
+                                        stop=0.2,
+                                        num=Nx),
+                            'y':
+                            sc.linspace(dim='y',
+                                        unit='m',
+                                        start=0.1,
+                                        stop=0.3,
+                                        num=Ny),
+                        })
+
+
+def test_rename_2_steps():
+    original = _make_xy()
+    graph = {'y4': 'y', 'y3': 'y2', 'y2': 'y'}
+    da = original.copy().transform_coords(['y3'],
+                                          graph=graph,
+                                          include_aliases=True)
+    assert da.dims == ['x', 'y3']
+    original = original.rename_dims({'y': 'y3'})
+    assert sc.identical(da.coords['y3'], original.coords['y'])
+    assert sc.identical(da.attrs['y2'], original.coords['y'])
+    assert sc.identical(da.attrs['y'], original.coords['y'])
+
+    da = original.copy().transform_coords(['y3'],
+                                          graph=graph,
+                                          include_aliases=False)
+    assert 'y3' in da.coords  # alias, but requested explicitly
+    assert 'y2' not in da.attrs  # alias for y => removed
+    assert 'y' in da.attrs
+
+
+def test_rename_multi_output():
+    original = _make_xy()
+    da = original.copy().transform_coords(['y3', 'y4'],
+                                          graph={
+                                              'y4': 'y',
+                                              'y3': 'y2',
+                                              'y2': 'y'
+                                          },
+                                          include_aliases=True)
+    assert da.dims == ['x', 'y']  # y4 also depends on y so dim not renamed
+    assert sc.identical(da.coords['y4'], original.coords['y'])
+    assert sc.identical(da.coords['y3'], original.coords['y'])
+    assert sc.identical(da.attrs['y2'], original.coords['y'])
+    assert sc.identical(da.attrs['y'], original.coords['y'])
+
+
+def ab(*, a, b):
+    return a + b
+
+
+def bc(*, b, c):
+    return b + c
+
+
+def test_avoid_consume_of_requested_outputs():
+    var = sc.arange(dim='a', start=0, stop=4)
+    original = sc.DataArray(data=var, coords={'a': var})
+
+    def split(*, a):
+        return {'b': a, 'c': a}
+
+    graph = {('b', 'c'): split, 'ab': ab}
+    da = original.transform_coords(['ab', 'b'], graph=graph)
+    assert 'b' in da.coords
```

Comment:
In many of the tests here, you are asserting that the correct dimensions are found in the resulting data array.
Should we also test that the values of the coords are correct, e.g. `a + b` has really been applied?

Reply:
Added more asserts along those lines.